VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    }
+    Foreach ($vCenter in $vCenterServer) {
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $vCenter.fqdn
+            'name'       = $vCenter.fqdn.split(".")[0]
+            'resourceId' = $vCenter.id
+            'type'       = "VCENTER"
+        }
+    }
+
+    # NSX V Manager
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management") -OR ($isLegacyWldnsxV))
+    {
+        $nsxvManager = Get-VCFNsxvManager | Where-Object { $_.domain.id -eq $domain.id }
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $nsxvManager.fqdn
+            'name'       = $nsxvManager.fqdn.split(".")[0]
+            'resourceId' = $nsxvManager.id
+            'type'       = "NSX_MANAGER"
+        }
+    }
+
+    #VRLI
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management"))
+    {
+        $logInsight = Get-VCFvRLI *>$null
+        If ($logInsight)
+        {
+            $resourcesObject += [pscustomobject]@{
+                'fqdn'       = ($logInsight.nodes | Where-Object {$_.type -eq "MASTER"}).fqdn
+                'name'       = (($logInsight.nodes | Where-Object {$_.type -eq "MASTER"}).fqdn).split(".")[0]
+                'resourceId' = $logInsight.id
+                'type'       = "VRLI"
+            }    
+        }
+    }
+    
+    #PSCs
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management"))
+    {
+        $pscs = Get-VCFPsc | Where-Object { $_.domain.id -eq $domain.id }
+        Foreach ($psc in $pscs) {
+            $resourcesObject += [pscustomobject]@{
+                'fqdn'       = $psc.fqdn
+                'name'       = $psc.fqdn.split(".")[0]
+                'resourceId' = $psc.id
+                'type'       = "PSC"
+            }
+        }  
+    }
+
+    #NSXT
+    If (([float]$sddcMgrVersion -ge [float]4) -OR ($isLegacyWldnsxT))
+    {
+        $nsxtManager = Get-VCFNsxtCluster | Where-Object { $_.domains.id -eq $domain.id }
+        $nsxtSans = @()
+        Foreach ($nodeFqdn in $nsxtManager.nodes.fqdn) 
+        {
+            $nsxtSans += $nodeFqdn
+        }
+
+        If ([float]$sddcMgrVersion -ge [float]4)
+        {
+            $nsxtSans += $nsxtManager.vipFqdn
+            $nsxtvip = $nsxtManager.vipfqdn
+        }
+        else
+        {
+            $nsxtSans += $nsxtManager.Fqdn
+            $nsxtvip = $nsxtManager.fqdn
+        }
+        Foreach ($nsxtManager in $nsxtManager) {
+            $resourcesObject += [pscustomobject]@{
+                'fqdn'       = $nsxtvip
+                'name'       = $nsxtvip.split(".")[0]
+                'resourceId' = $nsxtManager.id
+                'sans'       = $nsxtSans
+                'type'       = "NSXT_MANAGER"
+            }
+        }
+
+    }
+    Return $resourcesObject
+}
+
+Function Invoke-GenerateCsr {
```

Comment:
We currently have `Request-EsxiCsr` for ESXi hosts which is very similar. I see the opportunity here to have a `Request-Csr` (_naming TDB_) that could have a switch for the scope (e.g. host vs wld) and then require specific parameters based on the switch.

Thoughts?

Reply:
Add cmdlet metadata.

Reply:
ignore for now but will create a new issue on refactoring request-esxicsr and request-vcfgeneratecsr into one single function.
.json files will be removed after cmdlets runs(131acd4bec5c68a803eb55ef8c6b8d2638b81d17) 
switch to retain JSON is ignore for now, but I will create a new issue on improving the function with a switch or through debugPreference.